lib/HLSL/HLOperations.cpp

```diff
@@ -510,6 +500,9 @@ Function *GetOrCreateHLFunction(Module &M, FunctionType *funcTy,
   }
   else {
     mangledNameStr << GetHLFullName(group, opcode);
+    // Need to add wave sensitivity to name to prevent clashes with non-wave intrinsic
+    if(attribs.hasAttribute(AttributeSet::FunctionIndex, HLWaveSensitive))
+        mangledNameStr << "waveOp";
```

Comment:
See WaveProtMatch.hlsl test below for justification for this.

Reply:
GetHLFullName ends with a "." The result of this is  "dx.hl.op.waveOp." 

Anything would be enough to differentiate these. I was patterning it after "binop"

Reply:
Ok, that sounds good.